lib/network_utils.ml

```diff
@@ -674,6 +674,26 @@ module Proc = struct
 		else
 			raise Not_found
 
+	let get_vlans ?(full_conf=false) () =
+		try
+			let vlan_config = String.split '\n' (Unixext.string_of_file "/proc/net/vlan/config") in
+			(* First 2 lines from config file are not needed *)
+			let vlan_config_list =
+				match vlan_config with
+				| a::b::l -> l
+				| _ -> []
+			in
+			let create_config_list lst =
+				if full_conf then
+					Scanf.sscanf lst "%s | %d | %s" (fun device vlan parent -> device, vlan, parent)
+				else
+					Scanf.sscanf lst "%s | %d | %s" (fun device vlan parent -> "", vlan, parent)
+			in
+			List.map (fun x -> create_config_list x) (List.tl (List.rev vlan_config_list))
```

Comment:
Why is the `List.tl` etc. needed here? To keep it simple, I would just try to parse all lines and ignore lines that cannot be parsed (and not even remove the first two lines). Also removing the `full_conf` parameter, something like:
```
let get_vlans () =
	try
		List.file_lines_fold (fun vlans line ->
			try
				let x = Scanf.sscanf line "%s | %d | %s" (fun device vlan parent -> device, vlan, parent)) in
				x :: vlans
			with _ ->
				vlans
		) [] "/proc/net/vlan/config"
	with e ->
		error "Error: could not read /proc/net/vlan/config";
		[]
```